mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -20,6 +22,7 @@ pub struct PartQueue<E: std::error::Error> {
     failed: bool,
     /// The total number of bytes sent to the underlying queue of `self.receiver`
     bytes_received: Arc<AtomicUsize>,
+    mem_limiter: Arc<MemoryLimiter<Client>>,
```

Comment:
won't be needed if we don't account memory used by backward seeks? (my reasoning here is that this should be negligible)